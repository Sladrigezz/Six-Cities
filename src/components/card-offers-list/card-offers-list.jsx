import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';

import {CardOffer} from '../card-offer/card-offer.jsx';
import withActiveItem from '../../hocs/with-active-item/with-active-item.js';

const CardOfferWrapped = withActiveItem(CardOffer);

export const CardOffersList = (props) => {
  const {offers} = props;

  return (
    offers.map((item, i) => {
      const uniqueName = `offer-${i + 1}`;
      return <CardOfferWrapped
        key={uniqueName}
        offer={item}
        offerName={uniqueName}
        id={item.id}
      />;
    })
  );
};

CardOffersList.propTypes = {
  offers: PropTypes.array,
};

export default connect(
  (state) => ({
    offers: state.data.filteredOffers,
  })
)(CardOffersList);
