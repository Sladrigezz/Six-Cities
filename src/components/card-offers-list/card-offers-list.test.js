import React from 'react';
import renderer from 'react-test-renderer';
import {Router} from 'react-router-dom';

import {CardOffersList} from './card-offers-list.jsx';
import history from '../../history/history.js';

it(`should match snapshot`, () => {
  const offers = [{
    name: `AAA`,
    price: 120,
    type: `AAA`,
    images: [`aaa`],
  }];

  const wrapper = renderer.create(
      <Router history={history}>
        <CardOffersList offers={offers} />
      </Router>
  ).toJSON();

  expect(wrapper).toMatchSnapshot();
});
