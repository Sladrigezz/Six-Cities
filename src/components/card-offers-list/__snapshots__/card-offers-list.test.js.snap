// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<article
  className="cities__place-card place-card"
>
  <div
    className="cities__image-wrapper place-card__image-wrapper"
  >
    <img
      alt="Place image"
      className="place-card__image"
      height="200"
      onMouseEnter={[Function]}
      src="aaa"
      width="260"
    />
  </div>
  <div
    className=" place-card__info"
  >
    <div
      className="place-card__price-wrapper"
    >
      <div
        className="place-card__price"
      >
        <b
          className="place-card__price-value"
        >
          €
          120
        </b>
        <span
          className="place-card__price-text"
        >
          / night
        </span>
      </div>
      <button
        className="place-card__bookmark-button place-card__bookmark-button button"
        onClick={[Function]}
        type="button"
      >
        <svg
          className="place-card__bookmark-icon"
          height="19"
          width="18"
        >
          <use
            xlinkHref="#icon-bookmark"
          />
        </svg>
        <span
          className="visually-hidden"
        >
          In bookmarks
        </span>
      </button>
    </div>
    <div
      className="place-card__rating rating"
    >
      <div
        className="place-card__stars rating__stars"
      >
        <span
          style={
            Object {
              "width": "NaN%",
            }
          }
        />
        <span
          className="visually-hidden"
        >
          Rating
        </span>
      </div>
    </div>
    <h2
      className="place-card__name"
    >
      <a
        href="/offer/undefined"
        onClick={[Function]}
      />
    </h2>
    <p
      className="place-card__type"
    >
      AAA
    </p>
  </div>
</article>
`;
